core/src/main/java/org/shadowmask/core/mask/rules/AESCipher.java

```
@@ -0,0 +1,134 @@
+        this.method = "AES";
+    }
+
+    /*********
+    static {
+      try {
+        keygen_ = KeyGenerator.getInstance(this.method);
+
+        keygen_.init(128);
+        skey_ = keygen_.generateKey();
+
+        rand.setSeed(seed);
+        rand.nextBytes(iv_);
+        ivspec_ = new IvParameterSpec(iv_);
+      } catch(NoSuchAlgorithmException e) {
+        e.printStackTrace();
+      }
+    }
+    ***********/
+
+    public String evaluate(String content, int mode, String key) {
+      if (content == null) return null;
+      if (key == null || key != null && (key.length() % 8) != 0) {
+        throw new RuntimeException("Please input correct encrypting/decrypting key!\n" +
+            "Key length is the times of 8!");
+      }
+      // validate key and geberate the secret key
+      try {
+        skey_spec_ = new SecretKeySpec(key.toString().getBytes("UTF-8"), this.method);
+      }catch(UnsupportedEncodingException e) {
+        e.printStackTrace();
+      }
+
+      String result = null;
+      switch(mode) {
+        case Cipher.ENCRYPT_MODE:
+          result = encrypt(content.getBytes());
+          System.out.println("Encrypted text is " + result.toString());
+          break;
+        case Cipher.DECRYPT_MODE:
+          result = decrypt(Base64.getDecoder().decode(content.toString()));
+          System.out.println("Decrypted text is " + result.toString());
+          break;
+        default:
+          throw new RuntimeException("Unknown mode! [1: encryption, 2: decryption]");
+      }
+      return result;
+    }
+
+
+    protected String encrypt(byte[] content) {
+      String result = null;
+      try {
+        Cipher aesCipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+        aesCipher.init(Cipher.ENCRYPT_MODE, skey_spec_, ivspec_);
+
+        result = Base64.getEncoder().encodeToString(aesCipher.doFinal(content));
+        return result;
+      } catch(NoSuchAlgorithmException e) {
+        e.printStackTrace();
```

1. should we throw the exception instead of just returning null here?
2. we should log the error message instead of printing it.